### PR TITLE
Fixed Duplicate code - 1

### PR DESCRIPTION
**Pull request by mihir1739**
Solving #1262 
Part - 1
I have solved the first using closure and a function. Please give me feedback so that I can finish the other parts. 
Thankyou !

### CODE REVIEW

File: src/codegen/strength_reduce/expression_values.rs [hunk 1]
```diff
@@ -472,7 +472,23 @@ fn multiply_values(
         })
         .collect()
 }
-
+fn get_value(
```
Comment (seanyoung) [inline src/codegen/strength_reduce/expression_values.rs:475]: `get_value` is not a good name

File: src/codegen/strength_reduce/expression_values.rs [hunk 2]
```diff
@@ -472,7 +472,23 @@ fn multiply_values(
...
+    _signed : bool,
+    _l :&Value,
+    _r :&Value,
```
Comment (seanyoung) [inline src/codegen/strength_reduce/expression_values.rs:478]: please don't start variable names with `_`. Also `left` and `right` is better than `l` and `r`.

**Review by seanyoung — CHANGES_REQUESTED** [review]
There are test failures - these need fixing. You can test them locally using `cargo test`.

Also note the `cargo fmt` issue in CI.

File: src/codegen/strength_reduce/expression_values.rs (hunk 2 above)
Comment (LucasSte) [reply src/codegen/strength_reduce/expression_values.rs:478]: In addition, please, rename `_signed` to `signed`. In Rust, parameters that start with an underscore are not used in the function.

**Comment by LucasSte** [comment]
This PR has had not activity for more than a month. Can we close it?

**Comment by LucasSte** [comment]
> This PR has had no activity for more than a month. Can we close it?

**Comment by LucasSte** [comment]
> This PR has had no activity for more than a month. Can we close it?

@mihir1739 If you plan to get this merged in reasonable time, please let us know.

**Comment by LucasSte** [comment]
I'm closing this PR due to the lack of activity. @mihir1739, feel free to open it again in case you want to continue working.

**Pull request closed without merging**